frontend/packages/ceph-storage-plugin/src/features.ts

```diff
@@ -20,19 +20,27 @@ export const detectIndependentMode: ActionFeatureFlagDetector = (dispatch) =>
     },
   );
 
-export const OCS_VERSION_4_5_FLAG = 'OCS_VERSION_4_5_FLAG';
+export const OCS_SNAPSHOT = 'OCS_SNAPSHOT';
 
-export const isOCS45AndAboveVersion = (subscription: K8sResourceKind): boolean => {
-  const version = subscription?.status?.currentCSV;
-  return version && version.includes('ocs-operator.v4') && version.split('.')[2] >= 5;
+export const isOCSSnapshotSupported = (csv: K8sResourceKind): boolean => {
+  const support = csv?.metadata?.annotations['cluster.ocs.openshift.io/supported'];
+  return support.indexOf('snapshot') !== -1;
 };
 
-export const detectOCSVersion45AndAbove: ActionFeatureFlagDetector = (dispatch) =>
+export const detectOCSSupportedFeatures: ActionFeatureFlagDetector = (dispatch) =>
   k8sGet(SubscriptionModel, 'ocs-subscription', CEPH_STORAGE_NAMESPACE).then(
-    (obj: K8sResourceKind) => dispatch(setFlag(OCS_VERSION_4_5_FLAG, isOCS45AndAboveVersion(obj))),
-    (err) => {
-      err?.response?.status === 404
-        ? dispatch(setFlag(OCS_VERSION_4_5_FLAG, false))
-        : handleError(err, OCS_VERSION_4_5_FLAG, dispatch, detectOCSVersion45AndAbove);
+    (subscription: K8sResourceKind) => {
+      k8sGet(
+        ClusterServiceVersionModel,
+        subscription?.status?.currentCSV,
+        CEPH_STORAGE_NAMESPACE,
+      ).then(
+        (obj: K8sResourceKind) => dispatch(setFlag(OCS_SNAPSHOT, isOCSSnapshotSupported(obj))),
```

Comment:
Loop over enum & set the flags.

Reply:
Will give an issue is you set a flags which is not even defined yet. Can't do this.

Reply:
That's why we will iter over the Flags we have in UI & see if the corresponding flag to feature is available in CSV. if not the value will be set to false.

I think I said on a high level.